test/dotnet.Tests/CommandTests/Run/RunFileTests.cs

```diff
@@ -1293,15 +1297,11 @@ Override targets which don't work with project files that are not present on dis
 
                     </Project>
 
-                    """,
-                Diagnostics =
-                [
-                    new()
-                    {
-                        Location = new FileLinePositionSpan(programPath, new LinePosition(0, 0), new LinePosition(1, 0)),
-                        Message = string.Format(CliCommandStrings.UnrecognizedDirective, "unknown", $"{programPath}:1"),
-                    },
-                ],
-            }, RunFileApiJsonSerializerContext.Default.RunApiOutput));
+                    """)}},"Diagnostics":
+                [{"Location":{
+                "Path":{{ToJson(programPath)}},
+                "Span":{"Start":{"Line":0,"Character":0},"End":{"Line":1,"Character":0}{{nop}}}{{nop}}},
```

Comment:
Oh, is this nop breaking-up the `}`s in order to keep distinct which ones are ending an interpolation and which ones are part of the content?

Reply:
The nop is to avoid compiler errors due to the closing brackets `}}` inside an interpolated string `$$"""`. (the errors are redundant IMO, the interpolated string could be compiled just fine even though it contains unmatched `}}`.) See also the doc comment at the `nop` constant.

Example:

```cs
var s1 = $$"""{"x":{{1}},"y":{"z":2}}"""; // error on the }}

var s2 = $$$"""{"x":{{{1}}},"y":{"z":2}}"""; // ok but interpolation holes need soo many brackets (even more when the JSON is more nested)

const string nop = "";

var s3 = $$"""{"x":{{1}},"y":{"z":2}{{nop}}}"""; // workaround
```